lib/segment/src/index/struct_payload_index.rs

```diff
@@ -241,16 +243,16 @@ impl PayloadIndex for StructPayloadIndex {
         self.config.indexed_fields.clone()
     }
 
-    fn set_indexed(&mut self, field: &PayloadKeyType) -> OperationResult<()> {
-        if !self.config.indexed_fields.contains(field) {
-            self.config.indexed_fields.push(field.clone());
+    fn set_indexed(&mut self, field: PayloadKeyTypeRef) -> OperationResult<()> {
+        if !self.config.indexed_fields.iter().any(|x| x == field) {
```

Comment:
same

Reply:
Agree, but there is an unfortunate problem that contains method on Vec<String> accepts only &String and not &str. I did not find a way to convert &str to &String without creating the String,  which is a copying operation. Do you have any idea? Performance wise contains doing exactly the same as the above code .

Reply:
Oh, I didn't realize it was actually a new type at first glance. In this case, readability is not so critical compared to consistency.